data_store/messaging/messaging.py

```diff
@@ -45,21 +45,30 @@ def remove_errors_already_caught_by_null_failure(error_messages: list[Message])
         msgs.BLANK_UNIT_OF_MEASUREMENT,
     ]
 
-    cells_covered_by_null_failures = [
-        (message.sheet, cell_index)
-        for message in error_messages
-        if message.description in null_descriptions
-        for cell_index in message.cell_indexes
-    ]
-
-    filtered_errors = [
-        message
-        for message in error_messages
-        if not any(
-            ((message.sheet, cell_index) in cells_covered_by_null_failures) for cell_index in message.cell_indexes
-        )
-        or message.description in null_descriptions
-    ]
+    cells_covered_by_null_failures = []
+    for message in error_messages:
+        if message.description not in null_descriptions:
+            continue
+
+        if message.cell_indexes is None:
+            continue
+
+        for cell_index in message.cell_indexes:
+            cells_covered_by_null_failures.append((message.sheet, cell_index))
+
+    filtered_errors = []
+    for message in error_messages:
+        if message.cell_indexes is None:
+            continue
+
+        is_covered_by_null_failure = False
+        for cell_index in message.cell_indexes:
+            if (message.sheet, cell_index) in cells_covered_by_null_failures:
+                is_covered_by_null_failure = True
+                break
```

Comment:
```suggestion
        is_covered_by_null_failure = any(
            (message.sheet, cell_index) in cells_covered_by_null_failures
            for cell_index in message.cell_indexes
        )
```

maybe

Reply:
I don't feel crazy strong about it. It was an `any()` before. I rewrote it because the line was too long. If the formatter would break it in multiple line's as it shows in your commit suggestion, I would add the suggestion.